src/sage/rings/polynomial/laurent_polynomial_mpair.pyx

```diff
@@ -1514,11 +1514,37 @@ cdef class LaurentPolynomial_mpair(LaurentPolynomial):
             x = 'x'
             i = 0
 
-        #construct ring if none
+        # construct ring if none
         if R is None:
             R = LaurentPolynomialRing(self.base_ring(), x)
 
-        return R({m[i]: c for m,c in self.dict().iteritems()})
+        return R({m[i]: c for m, c in self.dict().iteritems()})
+
+    def polynomial_construction(self):
```

Comment:
This is already implemented in the (hidden) method of `_fraction_pair()` (although it is $f = p / v$ rather than $f = p \cdot v$, but that is not used in your code). So you don't need to implement this. Sorry for not noticing earlier.

Reply:
It is not completely the same thing, though if the intermediate use of `_fraction_pair()` make it faster the code could be changed. I want the polynomial $p$ to contain no coordinate as factor.

Reply:
Ah, I see. The method name is misleading. Really the thing you want to extract is `self._poly` of the Laurent polynomial after running `_normalize()` on it. I think something like this is reasonable:
```python
    def monomial_factorization(self):
        """
        Factor ``self`` into a polynomial and a monomial.
        """
        self._normalize()
        ring = self._parent._R
        g = ring.gens()
        mon = ring.prod(g[i] ** j for i, j in enumerate(self._mon))
        return (self._poly, mon)
```
The doctest should also have a case using only `x` and also with a monomial with positive and mixed exponents.

Reply:
I follwed your suggestion except for the change of name, anyway opened to think about. The reason of the name `polynomial_construction` is because it is the name of the method for one variable Laurent polynomial in `laurent_polynomial.pyx`; in the method `characteristic_varieties` the number of variables of the Laurent polynomial can be one.

Reply:
Copying a bad name does not make it a good name. For the univariate case, that is explicitly how the Laurent polynomial is constructed too. This is not the case for the multivariate. On the contrary, you should make an alias for the univariate case.

Reply:
Done